fcd/codegen/translation_context_remill.cpp

```diff
@@ -0,0 +1,575 @@
+          blocks_to_lift.push(inst.branch_taken_pc);
+        break;
+
+      default:
+        break;
+    }
+  }
+  return block;
+}
+
+remill::Instruction &RemillTranslationContext::LiftInst(llvm::BasicBlock *block,
+                                                        uint64_t addr) {
+  auto &inst = GetOrDecodeInst(addr);
+
+  CHECK(remill::kLiftedInstruction == lifter->LiftIntoBlock(inst, block))
+      << "Can't lift instruction " << inst.Serialize() << " at " << std::hex
+      << addr << std::dec;
+
+  auto func = block->getParent();
+  llvm::IRBuilder<> builder(block);
+  switch (inst.category) {
+    case remill::Instruction::kCategoryConditionalBranch: {
+      auto true_block = GetOrCreateBlock(inst.branch_taken_pc, func);
+      auto false_block = GetOrCreateBlock(inst.branch_not_taken_pc, func);
+      auto cond = remill::LoadBranchTaken(block);
+      builder.CreateCondBr(cond, true_block, false_block);
+    } break;
+
+    case remill::Instruction::kCategoryDirectJump: {
+      auto iter = functions.find(inst.branch_taken_pc);
+      if (iter != functions.end()) {
+        remill::AddTerminatingTailCall(block, iter->second);
+      } else {
+        auto target = GetOrCreateBlock(inst.branch_taken_pc, func);
+        builder.CreateBr(target);
+      }
+    } break;
+
+    case remill::Instruction::kCategoryIndirectJump:
+      remill::AddTerminatingTailCall(block, intrinsics->jump);
+      break;
+
+    case remill::Instruction::kCategoryDirectFunctionCall: {
+      if (inst.branch_taken_pc != inst.next_pc) {
+        auto target = DeclareFunction(inst.branch_taken_pc);
+        auto args = remill::LiftedFunctionArgs(block);
+        auto call = builder.CreateCall(target, args);
+        call->setCallingConv(target->getCallingConv());
+        auto mem_ptr = remill::LoadMemoryPointerRef(block);
+        builder.CreateStore(call, mem_ptr);
+      } else {
+        LOG(WARNING) << "Not adding a subroutine self-call at " << std::hex
+                     << inst.pc << std::dec;
+      }
+    } break;
+
+    case remill::Instruction::kCategoryIndirectFunctionCall: {
+      auto target = intrinsics->function_call;
+      auto args = remill::LiftedFunctionArgs(block);
+      auto call = builder.CreateCall(target, args);
+      call->setCallingConv(target->getCallingConv());
+      auto mem_ptr = remill::LoadMemoryPointerRef(block);
+      builder.CreateStore(call, mem_ptr);
+    } break;
+
+    case remill::Instruction::kCategoryFunctionReturn:
+      builder.CreateRet(remill::LoadMemoryPointer(block));
+      break;
+
+    case remill::Instruction::kCategoryError:
+      remill::AddTerminatingTailCall(block, intrinsics->error);
+      break;
+
+    default:
+      break;
+  }
+  return inst;
+}
+
+const StubInfo *RemillTranslationContext::GetStubInfo(
+    llvm::Function *func) const {
+
+  if (func->isDeclaration()) {
+    return nullptr;
+  }
+  
+  llvm::Value *read = nullptr;
```

Comment:
Use a different variable name that does ln't shadow the POSIX read function.

Reply:
Addressed in ab7c324